### ISSUE TITLE
weird behaviour with tasks

### ISSUE THREAD

**Issue by descartin** [signalwire/freeswitch#575]
Dear all

we are having some freeswitch instances (currently on both 1.6.19 running in ubuntu 16 and 1.8.2 running in debian9) version, which are working fine.
In the setup, we use the scheduler for the sched_hangup and for a sched_api to trigger an application uuid_ring_ready on the uuid channel.
in lua script
**session:execute("sched_hangup","+" .. duration)**
**api:execute("sched_api","+" .. duration .. " none uuid_ring_ready " .. uuid)**

As I say, it works fine with about 4000-5000 concurrent calls and around 100 sps, but from time to time, we see that of the number of freeswitch tasks starts to increase until a moment when when executing the scheduler the freeswitch seems to freeze and block the event handling for a few moments (secs).
So we end by having much more tasks than the active sessions, since seems the freeswitch has not been able top delete the tasks once they are done.
Some other few times, we have a similar behaviour, seems the scheduler call makes the freeswich to "freeze" for a few secs, and we see that the number of tasks is flat.

On the first image you can see 2 freeswitch tasks, the blue line is ok, and the yellow one seems to not be able to free all the tasks scheduled
![image](https://user-images.githubusercontent.com/7767994/80019244-7f7db000-84d7-11ea-8826-5b0d028c907a.png)

On the other case, we see that the number of tasks seems to be constant while calls are flowing normally
![image](https://user-images.githubusercontent.com/7767994/80019361-ae942180-84d7-11ea-91b5-4c0151f7bcd1.png)

we have checked the servers (it happens from time to time in any of them) and we cannot find anything which make us think about any root cause.
By resetting the freeswitch the issue is solved.
We have tried to make some adjustments to forced a sched_del when ending a session to "force" a deletion of the particular uuid tasks, but we dont know if that makes sense or not.

Have you ever seen or somebody else reported a behaviour like this?

thanks a lot and regards
david

**Comment by mjerris** [comment]
what do you mean by "tasks" ?

**Comment by descartin** [comment]
hello Michael

i'm referring to tasks of the sched_api

for example 
```
task_id,task_desc,task_group,task_runtime,task_sql_manager,hostname
1,heartbeat,core,1587589873,0,mia-minos-1.bts.io
2,check_ip,core,1587589913,0,mia-minos-1.bts.io
9696,sched_api_function,1587468821-131324336@176.74.219.11,1587468851,0,mia-minos-1.bts.io
2015189,switch_ivr_schedule_hangup,58q2c2nlo-3243206033@65.198.234.193,1587590660,0,mia-minos-1.bts.io
2079400,switch_ivr_schedule_hangup,69fY9565517220eTbgGhEfFpHhA0d@BC00.CHMSS318.TMOBILEUSA.COM,1587592568,0,mia-minos-1.bts.io
2175092,switch_ivr_schedule_hangup,WqfU5563918220V1biGhEfFePcE0h@BC00.ATMSS344.TMOBILEUSA.COM,1587595207,0,mia-minos-1.bts.io
```

best regards
david

**Comment by mjerris** [comment]
i know we have had issues with transactions failing that have been fixed over the years.  with code that very old it’s probably one of those.  you’d have to find related patches or upgrade

**Comment by descartin** [comment]
thanks a lot Mike, the versions we use are for 1.6.19 a commit from Jul 13 2017 and for 1.8.2 a commit from Sep 26 2018.
Looking at the commits, I see this one that could be related  

```
commit 6f2466eb35346c849144f01ce198bd617f0c4bef
Author: Chris Rienzo <chris@signalwire.com>
Date:   Mon Apr 22 17:10:46 2019 -0400

    FS-11801 [core] Update scheduler to allow destruction of running task
```

I see no more commits related to task or sched subjects since ~2015.

that could make sense maybe, am i right?
But seems that commit is for 1.10 version, not 1.8, am it right here?

best regards
david

**Comment by descartin** [comment]
hello Mike

ok then please take a look at issue https://github.com/signalwire/freeswitch/issues/577
i tried to install 1.10.2 and seems i got a limit on the number of max sessions.

Besides, i would like you to understand that this issue is not easy to duplicate since affects live production systems and we don't know which scenario creates it. 
I'm trying to provide you all the information i can, i would need you to at least take a look and consider the issue, the versions affected are 1.6.19 Jul 13th 2017 commit and for 1.8.2 a Sep 26th 2018 commit. And the patch i think can be related to this is from Arpil 2019. I don't see any other commit related to tasks, scheduler, etc. Maybe there is but i could not see it.

We already tried a patch from 1.10 related to tasks deletion, but in the case issue of 577 was solved and we moved our production systems to 1.10, maybe when we finish the deployment the version installed is also EOL, so we will be again in the same situation.

regards
david
